packages/credential-router/src/action-handler.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  IAgentPlugin,
+  ICreateVerifiableCredentialArgs,
+  ICreateVerifiablePresentationArgs,
+  ICredentialPlugin,
+  IssuerAgentContext,
+  IVerifyCredentialArgs,
+  IVerifyPresentationArgs,
+  IVerifyResult,
+  ProofFormat,
+  VerifiableCredential,
+  VerifiablePresentation,
+  VerifierAgentContext,
+} from '@vckit/core-types';
+
+import schema from '@vckit/core-types/build/plugin.schema.json' assert { type: 'json' };
+
+import Debug from 'debug';
+
+interface ICredentialRoutingConfig {
+  proofFormat: ProofFormat,
+
+}
+
+const debug = Debug('vckit:router:action-handler');
+
+/**
+ * A Veramo plugin that implements the {@link @veramo/core-types#ICredentialPlugin | ICredentialPlugin} methods.
+ *
+ * @public
+ */
+export class CredentialRouterPlugin implements IAgentPlugin {
+  readonly methods: ICredentialPlugin;
+  private credentialRouterConfig: any
+  readonly schema = {
+    components: {
+      schemas: {
+        ...schema.ICredentialIssuer.components.schemas,
+        ...schema.ICredentialVerifier.components.schemas,
+      },
+      methods: {
+        ...schema.ICredentialIssuer.components.methods,
+        ...schema.ICredentialVerifier.components.methods,
+      },
+    },
+  };
+
+  constructor(options: { routingConfig: any }) {
+    this.credentialRouterConfig = options.routingConfig
+    // @ts-ignore FIXME
+    this.methods = {
+      routerCreateVerifiablePresentation:
+        this.routerCreateVerifiablePresentation.bind(this),
+      routerCreateVerifiableCredential: this.routerCreateVerifiableCredential.bind(this),
+      verifyCredential: this.routerVerifyCredential.bind(this),
+      verifyPresentation: this.routerVerifyPresentation.bind(this),
+    };
+  }
+
+  getAgentMethod(
+    args: ICreateVerifiableCredentialArgs | ICreateVerifiablePresentationArgs | IVerifyCredentialArgs | IVerifyPresentationArgs,
+    context: IssuerAgentContext | VerifierAgentContext,
+    operation: string
+    ){
+
+      try {
+        const { proofFormat } = args;
```

Comment:
It seems like the args parameter in the `credential-w3c` for `verifyCredential` and `verifyPresentation` don't include the `proofFormat`, they will call `detectDocumentType(credential)` function to get the format type.  So, just to confirm, should we make `proofFormat` a mandatory field?